tasks/main.yml

```diff
@@ -186,53 +198,67 @@
     name: chronyd
     state: stopped
     enabled: false
-  when: timesync_mode != 1 or timesync_ntp_provider != 'chrony'
+  when:
+    - timesync_mode != 1 or timesync_ntp_provider != 'chrony'
+    - "'chrony' in _timesync_package_facts.ansible_facts.packages"
   ignore_errors: true
 
 - name: Disable ntpd
   service:
     name: ntpd
     state: stopped
     enabled: false
-  when: timesync_mode != 1 or timesync_ntp_provider != 'ntp'
+  when:
+    - timesync_mode != 1 or timesync_ntp_provider != 'ntp'
+    - "'ntp' in _timesync_package_facts.ansible_facts.packages"
   ignore_errors: true
 
 - name: Disable ntpdate
   service:
     name: ntpdate
     state: stopped
     enabled: false
```

Comment:
It might just work in check mode, and even if it does not, ignore_errors below should make the "when" unneeded.

Reply:
Correct, fixed.